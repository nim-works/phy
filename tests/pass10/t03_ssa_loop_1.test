(TypeDefs
  (Int 8)
  (ProcTy (Void)))
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1)
    (Locals (Type 0))
    (Stmts
      (Loop
        (Stmts
          (Asgn (Local 0) (IntVal 200))
          (If (IntVal 1)
            (Break 1))))
      (Drop (Copy (Local 0)))
      (Return))))
