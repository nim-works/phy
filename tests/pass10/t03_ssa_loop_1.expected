(TypeDefs
  (Int 8)
  (ProcTy (Void)))
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1)
    (Continuations
      (Continuation (Params) (Locals)
        (Continue 1 (List)))
      (Continuation (Params) (Locals)
        (Continue 2 (IntVal 200) (List)))
      (Continuation
        (Params (Type 0))
        (Locals)
        (SelectBool (IntVal 1)
          (Continue 4 (List))
          (Continue 3
            (List
              (Move (Local 0))))))
      (Continuation
        (Params (Type 0))
        (Locals)
        (Continue 5
          (List
            (Move (Local 0)))))
      (Continuation (Params) (Locals)
        (Loop 1 (List)))
      (Continuation
        (Params (Type 0))
        (Locals)
        (Drop
          (Copy (Local 0)))
        (Continue 6 (List)))
      (Continuation (Params)))))
